docs/calculating-effective-entry-date.md

```diff
@@ -0,0 +1,163 @@
+
+## 5. Effective Entry Date - Edge Cases
+
+### ENR Entries
+* For Automated Enrollment Entries (ENR), the Effective Entry Date field must be **space-filled**
+* ENR entries cannot be Same Day Entries
+
+### Notifications of Change
+* The ACH Operator does **not edit** the Effective Entry Date field for Notifications of Change
+
+### Return Entries
+* The ACH Operator may verify that the Effective Entry Date is properly formatted and may replace the field's existing content with the current processing date if the Effective Entry Date is invalid
+
+# Calculating Effective Entry Dates Using moov-io/base Functions
+
+## Overview
+
+The moov-io/base package provides functionality for calculating valid Effective Entry Dates when creating ACH batch headers, ensuring compliance with NACHA rules and Federal Reserve processing schedules.
+
+## Creating Batch Headers
+
+The NewBatchHeader function creates a BatchHeader object that requires several key fields to be populated according to NACHA rules:
+
+```go
+bh := NewBatchHeader()
+bh.ServiceClassCode = "220"                  // 220 for credits only
+bh.StandardEntryClassCode = PPD              // PPD, CCD, WEB, etc.
+bh.CompanyName = "ACME Corporation"          // Your company name
+bh.CompanyIdentification = "121042882"       // Your company ID
+bh.CompanyEntryDescription = "PAYROLL"       // Transaction description
+bh.ODFIIdentification = "12104288"           // Your bank's routing number
+
+bh.EffectiveEntryDate = "240102"             // YYMMDD format
+```
+
+## Banking Day Functions
+
+The moov-io/base Time package provides these essential functions for determining valid banking days:
+
+* `AddBankingTime(hours, minutes, seconds int)`: Increments time by specified duration while respecting banking hours (9am-5pm ET)
+* `AddBankingDay(d int)`: Adds the specified number of valid banking days to a date
+* `IsBankingDay()`: Validates if a date is a banking day by checking both holidays and weekends
+* `IsBusinessDay()`: Determines if a date is a standard business day
+* `IsHoliday()`: Checks if a date falls on a Federal Reserve holiday
+* `IsWeekend()`: Verifies if a date falls on Saturday or Sunday
+
+## Transmission Windows
+
+The package handles Same-Day ACH transmission windows with time-based validations. Here's how to implement the cutoff time checks:
+
+```go
+now := base.Now()
+// Create cutoff times for each window
+firstCutoff := now.AddBankingTime(10, 30, 0)    // 10:30 AM ET
+secondCutoff := now.AddBankingTime(14, 45, 0)   // 2:45 PM ET
+thirdCutoff := now.AddBankingTime(16, 45, 0)    // 4:45 PM ET
+
+if now.Before(firstCutoff.Time) {
+   // In first window
+   bh.EffectiveEntryDate = now.Format("060102")
+} else if now.Before(secondCutoff.Time) {
+   // First window closed, in second window
+   bh.EffectiveEntryDate = now.Format("060102")
+} else if now.Before(thirdCutoff.Time) {
+   // Second window closed, in third window
+   bh.EffectiveEntryDate = now.Format("060102")
```

Comment:
If the time is before the second or third cutoff it's before the first, so we could simplify this logic. 

Also, it would be worth adding a comment that in real world scenarios you typically want to have an internal cutoff prior to the Fed's cutoffs so your files have time to get uploaded before the Fed cutoff.

Reply:
This could all be simplified down to a single if where we check if the third window is before now, which means we need to .AddBankingDay(1)

Reply:
Simplified this down to a single check against the thrid (final) cutoff window.